### ISSUE TITLE
Mac M1 -> Ubuntu 24.10 (wayland) clipboard issue

### ISSUE THREAD

**Issue by emircanerkul** [deskflow/deskflow#7818]
### Project

Deskflow

### Deskflow version number

1.17.0.149 (077412f7)

### Build type

Deskflow package (downloaded from Deskflow)

### Operating systems (OS)

- [ ] Windows
- [X] macOS
- [ ] Linux (X11)
- [X] Linux (Wayland)
- [ ] BSD-derived
- [ ] Other (please specify)

### Wayland on Linux

- [X] I have reviewed the Wayland known issues and my issue is new
- [ ] I am not using Wayland on Linux

### Signing on macOS

- [X] I have authorized the app to run on my Mac
- [ ] I am not using macOS

### Continuous build

- [X] I have tried the latest continuous build and the issue persists
- [ ] I am unable to try the latest continuous build

### OS versions/distros

- Ubuntu 24.10
- Mac M1 ARM 15.0.1 (24A348)

### Deskflow configuration

- sa

### Additional information

I can see multiple log when i copy smth

[2024-10-28T07:03:57] INFO: screen "Emircans-MacBook-Air.local" grabbed clipboard 0 from "Emircans-MacBook-Air.local"
[2024-10-28T07:03:57] INFO: screen "Emircans-MacBook-Air.local" grabbed clipboard 1 from "Emircans-MacBook-Air.local"

**Comment by sithlord48** [comment]
Known issue: https://github.com/deskflow/deskflow/discussions/7499 , but since you checked you read that discussion you know that already

**Comment by nbolton** [comment]
> Known issue: #7499 , but since you checked " I have reviewed the Wayland known issues and my issue is new" you know that already...

Maybe he misread :smile: 

Re-surfacing for the next person:
> Copy and paste doesn't work on Wayland (https://github.com/flatpak/xdg-desktop-portal/issues/1434)

**Comment by tmountain** [comment]
Copy and paste works for me with Barrier + Wayland... curious how it works there but not with Deskflow.

**Comment by sithlord48** [comment]
Odd as barrier does not support Wayland or any of the portals. 

Tldr the portals we have to use on wayland do not currently provide access to the clipboard.

**Comment by tmountain** [comment]
Yes, I understand that the current situation is that deskflow does not work with wayland. This said, I am able to copy and paste if I run deskflow on my M1 mac and then run barrier on my Linux workstation. This isn't great though, as barrier has issues (bugs) that prevent me from using it in a productive fashion. Just wondering how this is possible and if there's a potential opportunity to make deskflow copy and paste work using whatever the underlying mechanism is. Currently, I can't really use deskflow, as copy and paste is essential.

**Comment by sithlord48** [comment]
Well that is very odd considering I am unable to get barrier to connect to anything in a wayland session..

**Comment by tmountain** [comment]
Happy to help diagnose or provide any underlying details if that is helpful. I can even make a video or something... Just let me know.

**Comment by sithlord48** [comment]
i would be interested to see the output of  `echo $XDG_SESSION_TYPE` and the barrier version as well as where you installed it from

**Comment by tmountain** [comment]
```
~ echo $XDG_SESSION_TYPE
wayland
```

Barrier version: 2.4.0-release-3e0d758b

Installed via Flathub (Fedora Silverblue)

**Comment by sithlord48** [comment]
but above it says your using `Ubuntu 24.10` so are you using that or fedora.. ?  have you tried the deskflow flatpak on fedora? it should use the same portal access barrier is using in that case. I am not aware of a barrier flatpak either. hmm ill have to go find that

**Comment by tmountain** [comment]
Oh, I'm sorry,  I am not the original reporter of the bug. I did not want to create a new ticket, as I know that the wayland copy/paste limitation is a known issue, and I didn't want to add more noise to your issues... I installed deskflow flatpak this morning from the latest version available. Here's the barrier version:

https://flathub.org/apps/com.github.debauchee.barrier

**Comment by nbolton** [comment]
Actually, you can use Deskflow in Xorg mode on Wayland. Strange, I know. It does so by making use of Xwayland (Xwayland is an Xorg emulation layer of Wayland). However, it behaves very strangely which is why we don't officially support it. When I was using it in this way, I was getting all kinds of weird bugs. This is unlikely to be something we'll ever support as we'll be focusing on fleshing out our Wayland support.

The Synergy team is currently in the process of getting quotes from contractors to add the missing clipboard support to `flatpak/xdg-desktop-portal` (and/or whatever else is missing to enable Wayland clipboard access). We plan to also use the same contractor to add the missing portal permission dialog persistence feature. If you know of any open source contractors with experience working on the OS itself, please let me know.

**Comment by sithlord48** [comment]
@tmountain  can you tell me what version of libportal is on your fedora machine ?

**Comment by tmountain** [comment]
I have `libportal.x86_64 version 0.8.1-2.fc41`. I also installed `libportal-gtk3` and `libportal-gtk4` with the same version.
